src/Components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,213 @@
+    setIsTitleEditing(isHasError);
+  }, [isHasError]);
+
+  useEffect(() => {
+    const handleKeyUp = (event: KeyboardEvent) => {
+      if (event.key === 'Escape') {
+        setIsTitleEditing(false);
+        document.removeEventListener('keyup', handleKeyUp);
+      }
+    };
+
+    if (isTitleEditing) {
+      document.addEventListener('keyup', handleKeyUp);
+    }
+  }, [isTitleEditing]);
+
+  const handleChange = () => {
+    onUpdateTodo(todo);
+  };
+
+  const handlerUpdateTodoTitle = async (updatedTodo: Todo) => {
+    onSetTodoIdsInLoading((prev: number[]) => [...prev, updatedTodo.id]);
+
+    if (!updatedTodo.title) {
+      deleteTodos(updatedTodo.id)
+        .then(() => {
+          onSetTodo(prevTodo =>
+            prevTodo.filter((curTodo: Todo) => curTodo.id !== updatedTodo.id),
+          );
+          setIsHasError(false);
+          onSetTodoEditingId(null);
+        })
+        .catch(() => {
+          onHandleErrorShow(ErrorMessage.DeletingError);
+          setIsHasError(true);
+        })
+        .finally(() => {
+          onSetTodoIdsInLoading(prev =>
+            prev.filter(curTodoId => curTodoId !== todo.id),
+          );
+        });
+
+      return;
+    }
+
+    onSetTodo(prevTodos =>
+      prevTodos.map(item => {
+        if (item.id === updatedTodo.id) {
+          return updatedTodo;
+        }
+
+        return item;
+      }),
+    );
+
+    patchTodo(updatedTodo.id, updatedTodo)
+      .then(() => {
+        setIsHasError(false);
+        onSetTodoEditingId(null);
+      })
+      .catch(() => {
+        onHandleErrorShow(ErrorMessage.UpdatingError);
+        setIsHasError(true);
+      })
+      .finally(() => {
+        onSetTodoIdsInLoading(prev =>
+          prev.filter(curTodoId => curTodoId !== updatedTodo.id),
+        );
+      });
+  };
+
+  const handleOnSubmit = async (event: React.FormEvent) => {
+    event.preventDefault();
+    onSetTodoEditingId(todo.id);
+    setEditedTitle(prev => prev.trim());
+
+    await handlerUpdateTodoTitle({
+      ...todo,
+      title: editedTitle.trim(),
+    } as Todo);
+
+    if (!isHasError) {
+      setIsTitleEditing(false);
+    }
+  };
+
+  const handleOnBlur = async () => {
+    if (todoEditingId !== null) {
+      setEditedTitle(prev => prev.trim());
+
+      await handlerUpdateTodoTitle({
+        ...todo,
+        title: editedTitle.trim(),
+      } as Todo);
+
+      if (!isHasError) {
+        setIsTitleEditing(false);
+      }
+    }
+  };
+
+  return (
+    <div
+      key={todo.id}
+      data-cy="Todo"
+      className={cn('todo', { completed: todo.completed }, 'item-enter-done')}
+    >
+      <label className="todo__status-label">
+        <input
+          aria-label="todo__status"
+          data-cy="TodoStatus"
+          type="checkbox"
+          className="todo__status"
+          checked={todo.completed}
+          onChange={handleChange}
+        />
+      </label>
+
+      {!isTitleEditing || todoEditingId !== todo.id ? (
```

Comment:
Let's create a separate variable for `!isTitleEditing || todoEditingId !== todo.id`